src/include/cart/api.h

```diff
@@ -1655,6 +1657,18 @@ crt_lm_attach(crt_group_t *tgt_grp, crt_lm_attach_cb_t completion_cb,
 int
 crt_proto_register(struct crt_proto_format *cpf);
 
+/**
+ * Unregister rpcs specified by base_opc and ver pairs.
+ *
+ * \param[in] base_opc         base_opc to unregister
+ * \param[in] ver              version to unregister
+ *
+ * \return                     DER_SPUCCESS on success, negative value
+ *                             on failure.
+ */
+int
+crt_proto_unregister(crt_opcode_t base_opc, uint32_t ver);
```

Comment:
But I'm not sure what the use-case is for a proto_unregister() function.  A version of a protocol is just that, a particular version so re-registering it later on with different values wouldn't make any sense, if there's a desire to shutdown or remove protocols I'd look at a enabled/disabled flag that could be set for either the protocol or protocol + version.

Reply:
This was added to provide previously existing ability to re-register rpcs in case someone was using it already. So instead of just registering on top of existing rpc, one would have to unregister and register new one if such thing is required for whatever reason

Reply:
Do we have any indication that it is required?

Reply:
Not direct no, but didnt seem it would hurt to have it

Reply:
I think this ability should be to correctly provide functionality of plug-ins. Server can load additional modules and dynamically extend functionality. Later this module should be able to be unloaded. So, unregistration is required functionality to avoid segmentation faults with dynamic modules.

Reply:
How is it safe to remove RPC definitions where RPCs could be arriving over the wire and being processed, or even if there are RPCs being sent from the local node that haven't completed?

The only safe way to unregister a protocol is if it doesn't define and server callbacks (so it's a client only protocol) or no RPCs are in flight from the local node.  These are both things that we could check for however without a use-case I would prefer we just didn't add this functionality until we know we would use it.

Reply:
Ok, will remove this functionality for now, and we can revisit this discussion once/if there is an actual need. Having RPCs on wire for old rpc is a good point as we dont do much protection against that